sockets.js

```diff
@@ -0,0 +1,188 @@
+var uuid = require('node-uuid'),
+    crypto = require('crypto');
+
+module.exports = function(server, io, matcher, config) {
+
+    io.sockets.on('connection', function(client) {
+
+        let { user_id, username } = client.handshake.session;
+        if (!username) {
+          client.emit('request username')
+        } else {
+          client.emit('recall username', username)
+          matcher.connect(client.id, username, user_id);
+        }
+
+        client.on("send message", function (data) {
+          client.emit("new message", `You said: ${data}`)
+          client.broadcast.to(matcher.getPartner(client.id)).emit("new message",
+            `${matcher.getUsername(client.id)} says: ${data}`
+            )
+
+        })
+
+        client.on("set user", ({ username, user_id }) => {
+          matcher.connect(client.id, username, user_id);
+          client.handshake.session.username = username;
+          client.handshake.session.save();
+          client.emit('recall username', username)
+        })
+
+        client.on("disconnect", () => {
+          matcher.disconnect(client.id);
+        })
+
+        client.on("hangup", () => {
+          matcher.hangup(client.id);
+        })
+
+        client.on("logout", () => {
+          delete client.handshake.session.user_id;
+          delete client.handshake.session.username;
+          client.handshake.session.save()
+          matcher.disconnect(client.id)
+        })
+
+        client.resources = {
+            screen: false,
+            video: true,
+            audio: false
+        };
+
+        // pass a message to another id
+        client.on('message', function(details) {
+            if (!details) return;
+
+            var otherClient = io.to(details.to);
+            if (!otherClient) return;
+
+            details.from = client.id;
+            otherClient.emit('message', details);
+        });
+
+        client.on('shareScreen', function() {
+            client.resources.screen = true;
+        });
+
+        client.on('unshareScreen', function(type) {
+            client.resources.screen = false;
+            removeFeed('screen');
+        });
+
+        client.on('join', join);
+
+        function removeFeed(type) {
+            if (client.room) {
+                io.sockets.in(client.room).emit('remove', {
+                    id: client.id,
+                    type: type
+                });
+                if (!type) {
+                    client.leave(client.room);
+                    client.room = undefined;
+                }
+            }
+        }
+
+        function join(name, cb) {
```

Comment:
what's the point of `cb`? any way we can convert this to use `Promise` instead to be cleaner?

Reply:
This code was taken directly from [Signalmaster](https://github.com/andyet/signalmaster) (MIT License), the signaling server designed to work with SimpleWebRTC